Tests failing due to zap asynchronous logging

We still have some tests failing because of zap logging, probably we need more fixes like #611.
https://circleci.com/gh/nspcc-dev/neo-go/15831
https://circleci.com/gh/nspcc-dev/neo-go/16963

```
panic: Log in goroutine after TestServerNotSendsVerack has completed

goroutine 86 [running]:
testing.(*common).logDepth(0xc00002fc20, 0xc0000c3020, 0x59, 0x3)
	/usr/local/go/src/testing/testing.go:680 +0x4d3
testing.(*common).log(...)
	/usr/local/go/src/testing/testing.go:662
testing.(*common).Logf(0xc00002fc20, 0xad2eb7, 0x2, 0xc000374f20, 0x1, 0x1)
	/usr/local/go/src/testing/testing.go:701 +0x7e
go.uber.org/zap/zaptest.testingWriter.Write(0xbc8b60, 0xc00002fc20, 0x0, 0xc0002f2800, 0x5a, 0x400, 0xc0002f2800, 0x9e0960, 0xc000374f10)
	/go/pkg/mod/go.uber.org/zap@v1.10.0/zaptest/logger.go:130 +0xfc
go.uber.org/zap/zapcore.(*ioCore).Write(0xc0000cb380, 0x0, 0xbf9e371677b2ab0f, 0xb4f1f303, 0x1031ca0, 0x0, 0x0, 0xade450, 0x12, 0x0, ...)
	/go/pkg/mod/go.uber.org/zap@v1.10.0/zapcore/core.go:90 +0x106
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0002f4000, 0xc000392d00, 0x2, 0x2)
	/go/pkg/mod/go.uber.org/zap@v1.10.0/zapcore/entry.go:215 +0x117
go.uber.org/zap.(*Logger).Info(0xc0000b0ba0, 0xade450, 0x12, 0xc000392d00, 0x2, 0x2)
	/go/pkg/mod/go.uber.org/zap@v1.10.0/logger.go:187 +0x7f
github.com/nspcc-dev/neo-go/pkg/network.(*Server).run(0xc0000eea00)
	/go/src/github.com/nspcc-dev/neo-go/pkg/network/server.go:243 +0xce5
created by github.com/nspcc-dev/neo-go/pkg/network.TestServerNotSendsVerack
	/go/src/github.com/nspcc-dev/neo-go/pkg/network/server_test.go:62 +0x1e5
FAIL	github.com/nspcc-dev/neo-go/pkg/network	3.292s
?   	github.com/nspcc-dev/neo-go/pkg/network/metrics	[no test files]
```
```
=== RUN   TestCreateBasicChain
==================
WARNING: DATA RACE
Read at 0x00c00019d943 by goroutine 11:
  testing.(*common).logDepth()
      /usr/local/go/src/testing/testing.go:621 +0x94
  testing.(*common).Logf()
      /usr/local/go/src/testing/testing.go:614 +0x90
  testing.(*T).Logf()
      <autogenerated>:1 +0x75
  go.uber.org/zap/zaptest.testingWriter.Write()
      /go/pkg/mod/go.uber.org/zap@v1.10.0/zaptest/logger.go:130 +0x11f
  go.uber.org/zap/zaptest.(*testingWriter).Write()
      <autogenerated>:1 +0xa9
  go.uber.org/zap/zapcore.(*ioCore).Write()
      /go/pkg/mod/go.uber.org/zap@v1.10.0/zapcore/core.go:90 +0x1c4
  go.uber.org/zap/zapcore.(*CheckedEntry).Write()
      /go/pkg/mod/go.uber.org/zap@v1.10.0/zapcore/entry.go:215 +0x1e7
  go.uber.org/zap.(*Logger).Info()
      /go/pkg/mod/go.uber.org/zap@v1.10.0/logger.go:187 +0x95
  github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).persist()
      /go/src/github.com/nspcc-dev/neo-go/pkg/core/blockchain.go:998 +0x6d4
  github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).Run.func2()
      /go/src/github.com/nspcc-dev/neo-go/pkg/core/blockchain.go:288 +0x53

Previous write at 0x00c00019d943 by goroutine 27:
  [failed to restore the stack]

Goroutine 11 (running) created at:
  github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).Run()
      /go/src/github.com/nspcc-dev/neo-go/pkg/core/blockchain.go:287 +0x237
==================
--- FAIL: TestCreateBasicChain (1.00s)
```
Affects both NEO 2 and NEO 3 branches.

--- a/pkg/core/helper_test.go
+++ b/pkg/core/helper_test.go
@@ -203,6 +203,7 @@ func TestCreateBasicChain(t *testing.T) {
 	var neoAmount = util.Fixed8FromInt64(99999000)
 	var neoRemainder = util.Fixed8FromInt64(100000000) - neoAmount
 	bc := newTestChain(t)
+	defer bc.Close()
 
 	// Move almost all NEO to one simple account.
 	txMoveNeo := transaction.NewContractTX()

--- a/pkg/network/server_test.go
+++ b/pkg/network/server_test.go
@@ -59,7 +59,16 @@ func TestServerNotSendsVerack(t *testing.T) {
 		p2 = newLocalPeer(t, s)
 	)
 	s.id = 1
-	go s.run()
+	finished := make(chan struct{})
+	go func() {
+		s.run()
+		close(finished)
+	}()
+	defer func() {
+		// close via quit as server was started via `run()`, not `Start()`
+		close(s.quit)
+		<-finished
+	}()
 
 	na, _ := net.ResolveTCPAddr("tcp", "0.0.0.0:3000")
 	p.netaddr = *na

